### ISSUE TITLE
mdecrypt_generic spam in nextcloud log

### ISSUE THREAD

**Issue by Hanzei** [nextcloud/server#3562]
### Steps to reproduce
1. Enable "External storage support"
2. Include SFTP server via GUI
3. Upload files via the nextcloud web interface into the ftp directory
4. Try to access these files
5. Nextcloud log gets spamed with PHP deprecated errors 

### Expected behaviour
Message should just apear once

### Actual behaviour
The logfile gets quite big. 6gb in ~ 6 hours

### Server configuration
**Operating system**:
CentOS 6
**Web server:**
Appache
**Database:**
MariaDB
**PHP version:**
7.1
**Nextcloud version:** (see Nextcloud admin page)
11.0.1
**Updated from an older Nextcloud/ownCloud or fresh install:**
Updated from Nextcloud 11.0
**Where did you install Nextcloud from:**
Tarball
**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - activity: 2.4.1
  - admin_audit: 1.1.0
  - calendar: 1.5.0
  - comments: 1.1.0
  - contacts: 1.5.3
  - dav: 1.1.1
  - direct_menu: 0.10.0
  - encryption: 1.4.1
  - external: true
  - federatedfilesharing: 1.1.1
  - files: 1.6.1
  - files_accesscontrol: 1.1.2
  - files_downloadactivity: 1.0.0
  - files_external: 1.1.2
  - files_markdown: 1.0.1
  - files_pdfviewer: 1.0.1
  - files_sharing: 1.1.1
  - files_texteditor: 2.2
  - files_trashbin: 1.1.0
  - files_versions: 1.4.0
  - files_videoplayer: 1.0.0
  - firstrunwizard: 2.0
  - gallery: 16.0.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - news: 10.1.0
  - nextcloud_announcements: 1.0
  - notes: 2.2.0
  - notifications: 1.0.1
  - password_policy: 1.1.0
  - provisioning_api: 1.1.0
  - rainloop: 4.26.1
  - serverinfo: 1.1.1
  - sharebymail: 1.0.1
  - survey_client: 0.1.5
  - tasks: 0.9.4
  - templateeditor: 0.2
  - theming: 1.1.1
  - twofactor_backupcodes: 1.0.0
  - twofactor_totp: 1.1.0
  - twofactor_u2f: 1.1.0
  - updatenotification: 1.1.1
  - workflowengine: 1.1.1
Disabled:
  - apporder
  - federation
  - files_automatedtagging
  - files_retention
  - keeweb
  - richdocuments
  - spreed
  - systemtags
  - user_external
  - user_ldap
  - user_saml
  - weather
```
</details>

**The content of config/config.php:**
<details>
<summary>Config report</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or 

Insert your config.php content here
(Without the database password, passwordsalt and secret)
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...
No
**Are you using encryption:** yes/no
yes
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
No

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
{"reqId":"86ugrdPi3rapyI8FFZn7","remoteAddr":"141.20.199.129","app":"PHP","message":"Function mdecrypt_generic() is deprecated at \/var\/www\/virtual\/minerva\/nextcloud.cjf-berlin.de\/3rdparty\/phpseclib\/phpseclib\/phpseclib\/Crypt\/Base.php#1131","level":3,"time":"2017-02-21T12:02:20+01:00","method":"GET","url":"\/index.php\/core\/preview.png?file=%2FBilder%2F2016%2F160117_Kr%C3%A4nzchen_Jahresanfangsessen%2FDSCF0929.JPG&c=58aa2aba0d313&x=32&y=32&forceIcon=0","user":"ben.schumacher","version":"11.0.1.2"}
{"reqId":"86ugrdPi3rapyI8FFZn7","remoteAddr":"141.20.199.129","app":"PHP","message":"Function mdecrypt_generic() is deprecated at \/var\/www\/virtual\/minerva\/nextcloud.cjf-berlin.de\/3rdparty\/phpseclib\/phpseclib\/phpseclib\/Crypt\/Base.php#1131","level":3,"time":"2017-02-21T12:02:20+01:00","method":"GET","url":"\/index.php\/core\/preview.png?file=%2FBilder%2F2016%2F160117_Kr%C3%A4nzchen_Jahresanfangsessen%2FDSCF0929.JPG&c=58aa2aba0d313&x=32&y=32&forceIcon=0","user":"ben.schumacher","version":"11.0.1.2"}
{"reqId":"XCZHhm618yKRbZNfTIpG","remoteAddr":"141.20.199.129","app":"PHP","message":"Function mdecrypt_generic() is deprecated at \/var\/www\/virtual\/minerva\/nextcloud.cjf-berlin.de\/3rdparty\/phpseclib\/phpseclib\/phpseclib\/Crypt\/Base.php#1131","level":3,"time":"2017-02-21T12:02:20+01:00","method":"GET","url":"\/index.php\/core\/preview.png?file=%2FBilder%2F2016%2F160117_Kr%C3%A4nzchen_Jahresanfangsessen%2FDSCF0933.JPG&c=58aa2ab9e1370&x=32&y=32&forceIcon=0","user":"ben.schumacher","version":"11.0.1.2"}
{"reqId":"XCZHhm618yKRbZNfTIpG","remoteAddr":"141.20.199.129","app":"PHP","message":"Function mdecrypt_generic() is deprecated at \/var\/www\/virtual\/minerva\/nextcloud.cjf-berlin.de\/3rdparty\/phpseclib\/phpseclib\/phpseclib\/Crypt\/Base.php#1131","level":3,"time":"2017-02-21T12:02:20+01:00","method":"GET","url":"\/index.php\/core\/preview.png?file=%2FBilder%2F2016%2F160117_Kr%C3%A4nzchen_Jahresanfangsessen%2FDSCF0933.JPG&c=58aa2ab9e1370&x=32&y=32&forceIcon=0","user":"ben.schumacher","version":"11.0.1.2"}
{"reqId":"XCZHhm618yKRbZNfTIpG","remoteAddr":"141.20.199.129","app":"PHP","message":"Function mcrypt_generic() is deprecated at \/var\/www\/virtual\/minerva\/nextcloud.cjf-berlin.de\/3rdparty\/phpseclib\/phpseclib\/phpseclib\/Crypt\/Base.php#837","level":3,"time":"2017-02-21T12:02:20+01:00","method":"GET","url":"\/index.php\/core\/preview.png?file=%2FBilder%2F2016%2F160117_Kr%C3%A4nzchen_Jahresanfangsessen%2FDSCF0933.JPG&c=58aa2ab9e1370&x=32&y=32&forceIcon=0","user":"ben.schumacher","version":"11.0.1.2"}
{"reqId":"H+l2GdFglxEW2Cq0QIjC","remoteAddr":"141.20.199.129","app":"PHP","message":"Function mcrypt_generic() is deprecated at \/var\/www\/virtual\/minerva\/nextcloud.cjf-berlin.de\/3rdparty\/phpseclib\/phpseclib\/phpseclib\/Crypt\/Base.php#837","level":3,"time":"2017-02-21T12:02:20+01:00","method":"GET","url":"\/index.php\/core\/preview.png?file=%2FBilder%2F2016%2F160117_Kr%C3%A4nzchen_Jahresanfangsessen%2FDSCF0938.JPG&c=58aa2aba0459d&x=32&y=32&forceIcon=0","user":"ben.schumacher","version":"11.0.1.2"}
{"reqId":"86ugrdPi3rapyI8FFZn7","remoteAddr":"141.20.199.129","app":"PHP","message":"Function mcrypt_generic() is deprecated at \/var\/www\/virtual\/minerva\/nextcloud.cjf-berlin.de\/3rdparty\/phpseclib\/phpseclib\/phpseclib\/Crypt\/Base.php#837","level":3,"time":"2017-02-21T12:02:20+01:00","method":"GET","url":"\/index.php\/core\/preview.png?file=%2FBilder%2F2016%2F160117_Kr%C3%A4nzchen_Jahresanfangsessen%2FDSCF0929.JPG&c=58aa2aba0d313&x=32&y=32&forceIcon=0","user":"ben.schumacher","version":"11.0.1.2"}
{"reqId":"XCZHhm618yKRbZNfTIpG","remoteAddr":"141.20.199.129","app":"PHP","message":"Function mdecrypt_generic() is deprecated at \/var\/www\/virtual\/minerva\/nextcloud.cjf-berlin.de\/3rdparty\/phpseclib\/phpseclib\/phpseclib\/Crypt\/Base.php#1131","level":3,"time":"2017-02-21T12:02:20+01:00","method":"GET","url":"\/index.php\/core\/preview.png?file=%2FBilder%2F2016%2F160117_Kr%C3%A4nzchen_Jahresanfangsessen%2FDSCF0933.JPG&c=58aa2ab9e1370&x=32&y=32&forceIcon=0","user":"ben.schumacher","version":"11.0.1.2"}
{"reqId":"XCZHhm618yKRbZNfTIpG","remoteAddr":"141.20.199.129","app":"PHP","message":"Function mdecrypt_generic() is deprecated at \/var\/www\/virtual\/minerva\/nextcloud.cjf-berlin.de\/3rdparty\/phpseclib\/phpseclib\/phpseclib\/Crypt\/Base.php#1131","level":3,"time":"2017-02-21T12:02:20+01:00","method":"GET","url":"\/index.php\/core\/preview.png?file=%2FBilder%2F2016%2F160117_Kr%C3%A4nzchen_Jahresanfangsessen%2FDSCF0933.JPG&c=58aa2ab9e1370&x=32&y=32&forceIcon=0","user":"ben.schumacher","version":"11.0.1.2"}
{"reqId":"XCZHhm618yKRbZNfTIpG","remoteAddr":"141.20.199.129","app":"PHP","message":"Function mcrypt_generic() is deprecated at \/var\/www\/virtual\/minerva\/nextcloud.cjf-berlin.de\/3rdparty\/phpseclib\/phpseclib\/phpseclib\/Crypt\/Base.php#837","level":3,"time":"2017-02-21T12:02:20+01:00","method":"GET","url":"\/index.php\/core\/preview.png?file=%2FBilder%2F2016%2F160117_Kr%C3%A4nzchen_Jahresanfangsessen%2FDSCF0933.JPG&c=58aa2ab9e1370&x=32&y=32&forceIcon=0","user":"ben.schumacher","version":"11.0.1.2"}

```
</details>

**Comment by Hanzei** [comment]
Update: Switching to PHP 7.1 fixes this problem

**Comment by nickvergessen** [comment]
phpseclib update required? cc @LukasReschke

**Comment by Hanzei** [comment]
phpseclib seems to fix the problem (see https://github.com/phpseclib/phpseclib/pull/1041)

**Comment by Hanzei** [comment]
What kind of info is needed?

**Comment by Hanzei** [comment]
Fixed by #4139
